src/scripts/mod.rs

```diff
@@ -0,0 +1,309 @@
+        }
+        let default_template: Template = Template::new(&final_call_format);
+        let exec_parts: ExecParts = ExecParts {
+            exec: first_argument,
+            ip: self.ip.to_string(),
+            port: ports_str,
+        };
+
+        let to_run: String = default_template.fill_with_struct(&exec_parts)?;
+
+        debug!("\nTo run {}", to_run);
+
+        let arguments: Vec<String> = to_run.split(" ").map(|arg| arg.to_string()).collect();
+
+        match execute_script(arguments) {
+            Ok(result) => return Ok(result),
+            Err(e) => return Err(e),
+        }
+    }
+}
+
+fn execute_script(mut arguments: Vec<String>) -> Result<String> {
+    debug!("\nArguments vec: {:?}", &arguments);
+    let process = Exec::cmd(&arguments.remove(0)).args(&arguments);
+    match process.capture() {
+        Ok(c) => {
+            let es = match c.exit_status {
+                ExitStatus::Exited(c) => c as i32,
+                ExitStatus::Signaled(c) => c as i32,
+                ExitStatus::Other(c) => c,
+                _ => -1,
+            };
+            if es != 0 {
+                return Err(anyhow!("Exit code = {}", es));
+            }
+            Ok(c.stdout_str())
+        }
+        Err(error) => {
+            debug!("Command error {}", error.to_string());
+            return Err(anyhow!(error.to_string()));
+        }
+    }
+}
+
+pub fn find_scripts(mut path: PathBuf) -> Result<Vec<PathBuf>> {
+    path.push(".rustscan_scripts");
+    // check if dir is available
+    if path.is_dir() {
+        debug!("Scripts folder found {}", &path.display());
+        let mut files_vec: Vec<PathBuf> = Vec::new();
+        for entry in fs::read_dir(path)? {
+            let entry = entry?;
+            files_vec.push(entry.path());
+        }
+        return Ok(files_vec);
+    } else {
+        return Err(anyhow!("cant find scripts folder"));
+        // println!("Do you want me to create the {} directory?", &scripts_dir.to_string_lossy());
+        // std::fs::create_dir(&scripts_dir).unwrap();
+    }
+}
+
+#[derive(Debug, Clone, Deserialize)]
+pub struct ScriptFile {
+    pub path: Option<PathBuf>,
+    pub tags: Option<Vec<String>>,
+    pub developer: Option<Vec<String>>,
+    pub execute_with: Option<String>,
+    pub execute_with_path: Option<PathBuf>,
+    pub join_ports: Option<String>,
+    pub call_format: Option<String>,
+    pub port: Option<String>,
+}
+
+pub fn parse_scripts(scripts: Vec<PathBuf>) -> Vec<ScriptFile> {
+    let mut parsed_scripts: Vec<ScriptFile> = Vec::with_capacity(scripts.len());
+    for script in scripts {
+        let real_path = script.clone();
+        let mut lines_buf = String::new();
+        if let Ok(file) = File::open(script) {
+            for line in io::BufReader::new(file).lines().skip(1) {
+                if let Ok(mut line) = line {
+                    if line.starts_with("#") && line.len() > 3 {
```

Comment:
I take it we are only doing Python / Shell scripts, assuming that the lines start with `#`?
It's good for an MVP, but we need to be aware of this bug in the future :)

Reply:
Python, perl, bash all uses `#` as comment, but we for sure can use an extra variable in the ScriptConfig to define that character. That way it will be completely dyunamic, the users could add own characters.